data_structures/huffman_coding.py

```diff
@@ -0,0 +1,102 @@
+# Huffman coding.
+
+
+from collections import Counter
+import heapq
+
+
+class HuffmanCoding:
```

Comment:
If you want to break logic up into encoding and decoding into its own classes, then HoffmanEncoder and HoffmanDecoder make sense.

Otherwise, call this a HoffmanTree and give it an encode and decode function.

Proper naming is important for understandability.